lib/Sema/SemaBounds.cpp

```diff
@@ -453,49 +529,57 @@ namespace {
         MemberExpr *ME = dyn_cast<MemberExpr>(E);
         FieldDecl *FD = dyn_cast<FieldDecl>(ME->getMemberDecl());
         if (!FD)
-          return CreateBoundsNone();
+          return CreateBoundsInferenceError();
         if (!SemaRef.CheckIsNonModifyingExpr(ME->getBase(),
                              Sema::NonModifiyingExprRequirement::NMER_Unknown,
                                            /*ReportError=*/false))
-          return CreateBoundsNone();
+          return CreateBoundsAllowedButUncomputable();
```

Comment:
Here is another place we now return `bounds(any)` instead of `bounds(none)`, because we want to allow these, but can't express the bounds correctly because if we did codegen again, we'd emit the modifying expression twice.